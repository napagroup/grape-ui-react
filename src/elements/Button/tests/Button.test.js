import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Button } from '..';

const emptyTheme = {};

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
describe('Button base', () => {
  it('should return a Button', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}><Button>Button Base</Button></ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an outlined Button', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button id="buttonOutline" outline>Button Outline</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an contained Button', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button contained id="buttonContained">Button Contained</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an contained raised Button', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button contained id="buttonContainedRaised" raised>Button Contained and Raised</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with disabled', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button disabled id="exampleColor">Disabled button</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with leading icon', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button leadingIcon="foo">Leading Icon on this button</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with loading indicator', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button showProgress>Loading...</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('Button Component base with style', () => {
  it('should return a Button with bg and color', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button bg="green" color="white" id="exampleColor">Happy St. Patrick&#39;s Day!</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with bg and color and bgActiveColor and bgHoverColor ', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button bg="green" bgActiveColor="green.light" bgHoverColor="green.dark" color="white" id="exampleColor">Happy St. Patrick&#39;s Day!</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with sm', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button id="exampleColor" sm>Do this</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with lg', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button id="exampleColor" lg>Do this</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with a zIndex when using a string', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button zIndex="500">Do this</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with a zIndex when using a number', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button zIndex={500}>Do this</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Link with href with button look', () => {
    const element = <ThemeProvider theme={emptyTheme}><Button href="https://www.google.com/" target="_blank">google</Button></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Link with to with button look', () => {
    const element = <ThemeProvider theme={emptyTheme}><Router><Button to="/404">404</Button></Router></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button object with mailto props filled out', () => {
    const component = renderer.create(
      <ThemeProvider theme={{}}>
        <Button
          emailHref={{
            bcc: 'email@napa.com',
            body: 'Body text for email.',
            cc: 'email@napa.com',
            subject: 'Subject line',
            to: 'email@napa.com',
          }}
        >
          Lorem Ipsum
        </Button>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
