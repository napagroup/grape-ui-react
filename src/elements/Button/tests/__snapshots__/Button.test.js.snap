// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component base with style should return a Button object with mailto props filled out 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<a
  className="c0"
  href="mailto:email@napa.com?bcc=email@napa.com&body=Body%20text%20for%20email.&cc=email@napa.com&subject=Subject%20line"
>
  <div
    className="c1 c2"
  >
    Lorem Ipsum
  </div>
</a>
`;

exports[`Button Component base with style should return a Button with a zIndex when using a number 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  z-index: 500;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  type="button"
>
  <div
    className="c1 c2"
  >
    Do this
  </div>
</button>
`;

exports[`Button Component base with style should return a Button with a zIndex when using a string 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  z-index: 500;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  type="button"
>
  <div
    className="c1 c2"
  >
    Do this
  </div>
</button>
`;

exports[`Button Component base with style should return a Button with bg and color 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: hsl(122.4,39.4%,49.2%);
  color: hsl(0,0%,98%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  <div
    className="c1 c2"
  >
    Happy St. Patrick's Day!
  </div>
</button>
`;

exports[`Button Component base with style should return a Button with bg and color and bgActiveColor and bgHoverColor  1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: hsl(122.4,39.4%,49.2%);
  color: hsl(0,0%,98%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(122.4,39.4%,29.2%);
}

.c0:active {
  background-color: hsl(122.4,39.4%,69.2%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  <div
    className="c1 c2"
  >
    Happy St. Patrick's Day!
  </div>
</button>
`;

exports[`Button Component base with style should return a Button with lg 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 8px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 8px;
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 8px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 8px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  <div
    className="c1 c2"
  >
    Do this
  </div>
</button>
`;

exports[`Button Component base with style should return a Button with sm 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 2px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 0;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 2px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  <div
    className="c1 c2"
  >
    Do this
  </div>
</button>
`;

exports[`Button Component base with style should return a Link with href with button look 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<a
  className="c0"
  href="https://www.google.com/"
  target="_blank"
>
  <div
    className="c1 c2"
  >
    google
  </div>
</a>
`;

exports[`Button Component base with style should return a Link with to with button look 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<a
  className="c0"
  href="/404"
  onClick={[Function]}
>
  <div
    className="c1 c2"
  >
    404
  </div>
</a>
`;

exports[`Button base should return a Button 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  type="button"
>
  <div
    className="c1 c2"
  >
    Button Base
  </div>
</button>
`;

exports[`Button base should return a Button with disabled 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  disabled={true}
  id="exampleColor"
  type="button"
>
  <div
    className="c1 c2"
  >
    Disabled button
  </div>
</button>
`;

exports[`Button base should return a Button with leading icon 1`] = `
.c3 {
  box-sizing: border-box;
  padding-right: 16px;
}

.c4 {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c2 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  type="button"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
    >
      foo
    </div>
    <div
      className="c2 c4"
    >
      Leading Icon on this button
    </div>
  </div>
</button>
`;

exports[`Button base should return a Button with loading indicator 1`] = `
.c6 {
  height: 24px;
  width: 24px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c7 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,75%);
}

.c5 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c8 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: kEfyFd;
  animation-name: kEfyFd;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c3 {
  box-sizing: border-box;
  padding-right: 16px;
}

.c4 {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c2 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  type="button"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <div
        className="c2 c4 c5"
      >
        <svg
          className="c6"
          display="block"
          height={24}
          overflow="hidden"
          width={24}
        >
          <circle
            circumference={62.83185307179586}
            className="c7"
            cx={12}
            cy={12}
            r={10}
            strokeWidth={4}
          />
          <circle
            circumference={62.83185307179586}
            className="c8"
            cx={12}
            cy={12}
            r={10}
            strokeDasharray={62.83185307179586}
            strokeWidth={4}
          />
        </svg>
      </div>
    </div>
    <div
      className="c2 c4"
    >
      Loading...
    </div>
  </div>
</button>
`;

exports[`Button base should return an contained Button 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="buttonContained"
  type="button"
>
  <div
    className="c1 c2"
  >
    Button Contained
  </div>
</button>
`;

exports[`Button base should return an contained raised Button 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 0;
  border-radius: 4px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.12),0 1px 5px 0 rgba(0,0,0,0.2);
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  box-shadow: 0 4px 5px 0 rgba(0,0,0,0.14),0 1px 10px 0 rgba(0,0,0,0.12),0 2px 4px -1px rgba(0,0,0,0.3);
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="buttonContainedRaised"
  type="button"
>
  <div
    className="c1 c2"
  >
    Button Contained and Raised
  </div>
</button>
`;

exports[`Button base should return an outlined Button 1`] = `
.c2 {
  box-sizing: border-box;
}

.c0 {
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c0 > .c1 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

<button
  className="c0"
  id="buttonOutline"
  type="button"
>
  <div
    className="c1 c2"
  >
    Button Outline
  </div>
</button>
`;
